easy-ui-react/src/SearchNav/SearchNav.tsx

```diff
@@ -103,7 +110,7 @@ export type SearchNavProps<T> = {
 export function SearchNav<T>(props: SearchNavProps<T>) {
   const { menuOverlayProps, ctaMenuSymbol, children } = props;
 
-  const context = useMemo(() => {
+  const { onlyLogoGroup, context } = useMemo(() => {
```

Comment:
**suggestion**: this memo function that derives the context is a little difficult to reason about. not a blocker, but if there's a way you can think of to make it easier to read, that would be helpful for future work

Reply:
thats a good point, i extracted the bits into separate functions and added a comment at the top of the memo function. hopefully that makes it a bit more readable.